tools/mirror/src/key_mapping.rs

```diff
@@ -0,0 +1,82 @@
+use hkdf::Hkdf;
+use near_crypto::{ED25519PublicKey, ED25519SecretKey, PublicKey, Secp256K1PublicKey, SecretKey};
+use sha2::Sha256;
+
+fn ed25519_map_secret(
+    buf: &mut [u8],
+    public: &ED25519PublicKey,
+    secret: Option<&[u8; crate::secret::SECRET_LEN]>,
+) {
+    match secret {
+        Some(secret) => {
+            let hk = Hkdf::<Sha256>::new(None, secret);
+            hk.expand(&public.0, buf).unwrap();
```

Comment:
I'm fairly sure we want `new(None, secret)`, since `secret` is the "input keying material" here: https://datatracker.ietf.org/doc/html/rfc5869#section-2.2, and then `public` is the app-specific info https://datatracker.ietf.org/doc/html/rfc5869#section-3.2.

I think it would make sense to maybe do `new(Some(genesis_block_hash), secret)` or something, but given that the target chain is basically always some test chain anyway, prob not that important